### DISCUSSION TITLE
Inconsistency in return type of model_dump_json vs TypeAdapter.dump_json

### DISCUSSION THREAD

**Discussion by bmarinb** [pydantic/pydantic#7882] [Question]
I wonder why model_dump_json returns a json string but TypeAdapter.dump_json returns bytes. I think this generates a consistency problem, since one expects the interface to behave similarly.

**Comment by cunnane** [comment]
This [is explained here](https://docs.pydantic.dev/latest/concepts/type_adapter/):

TypeAdapter's dump_json methods returns a bytes object, unlike the corresponding method for BaseModel, model_dump_json, which returns a str. The reason for this discrepancy is that in V1, model dumping returned a str type, so this behavior is retained in V2 for backwards compatibility. For the BaseModel case, bytes are coerced to str types, but bytes are often the desired end type. Hence, for the new TypeAdapter class in V2, the return type is simply bytes, which can easily be coerced to a str type if desired.
